Cargo.toml

```diff
@@ -162,6 +162,43 @@ default-members = [
 ]
 resolver = "2"
 
+#
+# Tree-wide lint configuration.
+# https://doc.rust-lang.org/stable/cargo/reference/manifest.html#the-lints-section
+#
+# For a list of Clippy lints, see
+# https://rust-lang.github.io/rust-clippy/master.
+#
+[workspace.lints.clippy]
+# Clippy's style nits are useful, but not worth keeping in CI.
+style = { level = "allow", priority = -1 }
```

Comment:
what does `priority = -1` do? i assume it allows this to be overridden, but why do we only need to set it here?

Reply:
Until `[lints]`, all methods for modifying lint levels had inherent ordering; either command line flags, or attributes in code. TOML tables are explicitly unordered, so Rust had to do something different here. [Per the RFC](https://rust-lang.github.io/rfcs/3389-manifest-lint.html#reference-level-explanation) any sorting beyond user-defined priorities is undefined ([more detail](https://rust-lang.github.io/rfcs/3389-manifest-lint.html#lint-precedence)); `style` has to come first because otherwise it would re-allow the `style` lints that happen to come before it once sorted.

The default `priority` for any key in this table is 0, so setting this to -1 means it is ordered before the rest of the keys.